manifests/install.pp

```diff
@@ -1,101 +1,54 @@
 # @api private
 #
-# @summary This private class installs the rundeck package and its dependencies.
+# @summary This class is called from rundeck for install.
 #
 class rundeck::install {
   assert_private()
 
-  $manage_repo        = $rundeck::manage_repo
-  $package_ensure     = $rundeck::package_ensure
-  $repo_yum_source    = $rundeck::repo_yum_source
-  $repo_yum_gpgkey    = $rundeck::repo_yum_gpgkey
-  $repo_apt_source    = $rundeck::repo_apt_source
-  $repo_apt_key_id    = $rundeck::repo_apt_key_id
-  $repo_apt_keyserver = $rundeck::repo_apt_keyserver
-  $user               = $rundeck::user
-  $group              = $rundeck::group
-  $manage_user        = $rundeck::manage_user
-  $manage_group       = $rundeck::manage_group
-  $user_id            = $rundeck::user_id
-  $group_id           = $rundeck::group_id
-
-  if $manage_group {
-    group { $group:
+  if $rundeck::manage_group {
+    group { $rundeck::group:
       ensure => present,
-      gid    => $group_id,
+      gid    => $rundeck::group_id,
       system => true,
     }
 
-    if $group != 'rundeck' {
+    if $rundeck::group != 'rundeck' {
       group { 'rundeck':
         ensure => absent,
       }
     }
   }
 
-  if $manage_user {
-    user { $user:
+  if $rundeck::manage_user {
+    user { $rundeck::user:
       ensure => present,
-      groups => [$group],
-      uid    => $user_id,
-      gid    => $group_id,
+      groups => [$rundeck::group],
+      uid    => $rundeck::user_id,
+      gid    => $rundeck::group_id,
       system => true,
-      before => File['/var/rundeck'],
     }
 
-    if $user != 'rundeck' {
+    if $rundeck::user != 'rundeck' {
       user { 'rundeck':
         ensure => absent,
       }
     }
   }
 
   case $facts['os']['family'] {
-    'RedHat': {
-      if $manage_repo {
-        yumrepo { 'rundeck':
-          baseurl       => $repo_yum_source,
-          descr         => 'rundeck repo',
-          enabled       => '1',
-          gpgcheck      => '0',
-          gpgkey        => $repo_yum_gpgkey,
-          repo_gpgcheck => '1',
-          priority      => '1',
-          before        => Package['rundeck'],
+    /RedHat|Debian/: {
+      if $rundeck::manage_repo {
+        $rundeck::repo_config.each() | String $_resource_type, Hash $_resources | {
+          if downcase($_resource_type) == 'apt::source' {
+            Class['Apt::Update'] -> Package['rundeck']
+          }
+          create_resources($_resource_type, $_resources, { 'before' => Package['rundeck'] })
```

Comment:
Simplify, and eliminate `create_resources`.

```suggestion
        $rundeck::repo_config.each |String $_apt_source, Hash $_attrs| {
          apt::source { $_apt_source:
            * => $_attrs,
            before => Package['rundeck'],
          }
```

Reply:
if @kenyon's suggestion about removing the `apt::source` and `yumrepo` from the hiera hash will be implemented then this part should be split for RH/Debian families again.

Reply:
Yep working on it will split them.

Reply:
Updated the loop not optimal in my opinion because you have some resource twice with the same value but i will try to optimise this tomorrow.
I wonder why the debian tests fail with: Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Error while evaluating a Function Call, key hash must contain at least an id entry (file: /etc/puppetlabs/code/environments/production/modules/apt/manifests/source.pp, line: 126, column: 9)

The syntax seems correct and valid according to the apt module

Reply:
Ah that's because the acceptance tests are installing puppetlabs-apt v9.1.0, but we need v9.2.0 for the modern keyrings feature: https://github.com/puppetlabs/puppetlabs-apt/pull/1150

Reply:
Ah ok that explains should i wait then and update acceptance tests with new apt release?

Reply:
The acceptance tests should use the new version automatically once it's released.

Reply:
Alright then i will wait for the apt PR and try to simplify the os family case structure and update the readme a bit more